app/src/main/java/fr/free/nrw/commons/depictions/DepictedImagesActivity.java

```diff
@@ -0,0 +1,396 @@
+import fr.free.nrw.commons.kvstore.JsonKvStore;
+import fr.free.nrw.commons.media.MediaClient;
+import fr.free.nrw.commons.media.MediaDetailPagerFragment;
+import fr.free.nrw.commons.theme.NavigationBaseActivity;
+import fr.free.nrw.commons.upload.structure.depicts.DepictedItem;
+import fr.free.nrw.commons.utils.NetworkUtils;
+import fr.free.nrw.commons.utils.ViewUtil;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.schedulers.Schedulers;
+import timber.log.Timber;
+
+import static android.view.View.GONE;
+import static android.view.View.VISIBLE;
+
+/**
+ * Actvity for shhowing image list after selected an item from SearchActivity In Explore
+ */
+
+public class DepictedImagesActivity extends NavigationBaseActivity implements FragmentManager.OnBackStackChangedListener, MediaDetailPagerFragment.MediaDetailProvider,
+        AdapterView.OnItemClickListener {
+
+    private static int TIMEOUT_SECONDS = 15;
+
+    private GridViewAdapter gridAdapter;
+    private FragmentManager supportFragmentManager;
+
+    private MediaDetailPagerFragment mediaDetails;
+    @BindView(R.id.mediaContainer)
+    FrameLayout mediaContainer;
+
+    @BindView(R.id.statusMessage)
+    TextView statusTextView;
+    @BindView(R.id.loadingImagesProgressBar)
+    ProgressBar progressBar;
+    @BindView(R.id.depicts_image_list)
+    GridView gridView;
+    @BindView(R.id.parentLayout)
+    RelativeLayout parentLayout;
+
+    private CompositeDisposable compositeDisposable = new CompositeDisposable();
+    private boolean hasMoreImages = true;
+    private boolean isLoading = true;
+    private String depictName = null;
+    private String entityId = null;
+    private List<Media> queryList = new ArrayList<>();
+    @Inject
+    DepictsClient depictsClient;
+    @Inject
+    MediaClient mediaClient;
+
+    private int mediaSize = 0;
+
+    @Inject
+    @Named("default_preferences")
+    JsonKvStore depictionKvStore;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_depict_detail);
+        ButterKnife.bind(this);
+        gridView.setOnItemClickListener((AdapterView.OnItemClickListener) this);
+        supportFragmentManager = getSupportFragmentManager();
+        supportFragmentManager.addOnBackStackChangedListener(this);
+        setPageTitle();
+        initDrawer();
+        forceInitBackButton();
+    }
+
+    /**
+     * Gets the passed depictsName from the intents and displays it as the page title
+     */
+    private void setPageTitle() {
+        if (getIntent() != null && getIntent().getStringExtra("depictsName") != null) {
+            setTitle(getIntent().getStringExtra("depictsName"));
+            entityId = getIntent().getStringExtra("entityId");
+            //resetQueryContinueValues(depictName);
+            initList();
```

Comment:
Should these be inside this function, setPageTitle ?

Reply:
I think they can be moved to onCreate.